### PR TITLE
Add basic TCPProxy support

### LINKED ISSUE #42: TCPProxy support does not work
Logged by @slguarderas in projectcontour/contour#2564:

When using the conversion tool, it failed to migrate the tcpproxy portion of the IngressRoute.

Here is the input:
```yaml
---
apiVersion: contour.heptio.com/v1beta1
kind: IngressRoute
metadata:
  name: tls-edge-terminate-prometheus-k8s-thanos
spec:
  virtualhost:
    fqdn: "tls-edge-terminate-prometheus-k8s-thanos{{ domain_apps }}"
    tls:
      secretName: "{{ default_contour_cert }}"
  tcpproxy:
    services:
    - name: prometheus-k8s-thanos
      port: 10901
```

Output:
```yaml
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: tls-edge-terminate-prometheus-k8s-thanos
spec:
  virtualhost:
    fqdn: tls-edge-terminate-prometheus-k8s-thanos{{ domain_apps }}
    tls:
      secretName: '{{ default_contour_cert }}'
```

**What did you expect to happen:**
The tcpproxy should have been included in the generated HttpProxy, as in:

```yaml
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: tls-edge-terminate-prometheus-k8s-thanos
spec:
  virtualhost:
    fqdn: tls-edge-terminate-prometheus-k8s-thanos{{ domain_apps }}
    tls:
      secretName: '{{ default_contour_cert }}'
  tcpproxy:
    services:
    - name: prometheus-k8s-thanos
      port: 10901
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Contour version:
- Kubernetes version: (use `kubectl version`):  v1.15.3+vmware.1
- Kubernetes installer & version:  VMware
- Cloud provider or hardware configuration:  
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by youngnick**
Fixes #42.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: internal/translator/translate.go [hunk 1]
```diff
@@ -203,6 +214,23 @@ func translateRoutes(irRoutes []irv1beta1.Route, routeLCP string) ([]hpv1.Route,
 	return routes, includes, warnings
 }
 
+func translateTCPProxy(irTCPProxy *irv1beta1.TCPProxy) *hpv1.TCPProxy {
+
+	proxy := &hpv1.TCPProxy{}
+
+	for _, irService := range irTCPProxy.Services {
```
Comment (jpeach) [inline internal/translator/translate.go:265]: If the Delegate field can't be translated, we should emit a warning.

File: internal/translator/translate.go [hunk 2]
```diff
@@ -203,6 +214,23 @@ func translateRoutes(irRoutes []irv1beta1.Route, routeLCP string) ([]hpv1.Route,
...
+
+		hpService := hpv1.Service{
+			Name:   irService.Name,
+			Port:   irService.Port,
+			Weight: irService.Weight,
```
Comment (jpeach) [inline internal/translator/translate.go]: What about the other service fields? Healthcheck, Strategy, UpstreamValidation? Could just hoist the code from `translateRoute` into something that translates a service and does all the fields ...

**Review by jpeach — CHANGES_REQUESTED** [review]

File: internal/translator/translate.go (hunk 2 above)
Comment (youngnick) [reply internal/translator/translate.go]: Those actually don't do anything for TCPProxy, afaict, so I left them out. I can do it that way if you think it's better though.

File: internal/translator/translate.go (hunk 1 above)
Comment (youngnick) [reply internal/translator/translate.go:265]: It's not valid to delegate tcpproxy IngressRoutes. I can check for that?

File: internal/translator/translate.go (hunk 2 above)
Comment (youngnick) [reply internal/translator/translate.go]: Now I think about it, I'll just do that, regardless of whether it does anything. Thanks.

File: internal/translator/translate.go (hunk 1 above)
Comment (youngnick) [reply internal/translator/translate.go:265]: Oh, I stand corrected. I'll add support for delegation.

Comment (jpeach) [reply internal/translator/translate.go:265]: I don't know, I was just looking at the API spec 🤷

File: internal/translator/translate.go [hunk 3]
```diff
@@ -123,50 +147,67 @@ func translateRoute(irRoute irv1beta1.Route, routeLCP string) (hpv1.Route, []str
...
+			Path:                    irService.HealthCheck.Path,
+			Host:                    irService.HealthCheck.Host,
+			TimeoutSeconds:          irService.HealthCheck.TimeoutSeconds,
+			UnhealthyThresholdCount: irService.HealthCheck.UnhealthyThresholdCount,
+			HealthyThresholdCount:   irService.HealthCheck.HealthyThresholdCount,
+		}
+	}
+
+	return service, healthcheckPolicy, lbpolicy
+}
```
Comment (jpeach) [inline internal/translator/translate.go:210]: Ni: newline between functions.

**Review by jpeach — APPROVED** [review]
LGTM

File: internal/translator/translate.go (hunk 3 above)
Comment (youngnick) [reply internal/translator/translate.go:210]: We are the knights who say Ni, and we agree.

(More seriously, no problem, fixed).

**Pull request merged**